src/mbgl/storage/resource.cpp

```diff
@@ -55,6 +89,8 @@ Resource Resource::tile(const std::string& urlTemplate,
                         int8_t z,
                         Necessity necessity) {
     bool supportsRatio = urlTemplate.find("{ratio}") != std::string::npos;
+    auto quadKey = getQuadKey(x, y, z);
```

Comment:
Move `getQuadKey` and `getTileBBox` into the lambda conditionals, so they are only called if needed.